ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/JaasAuthProvider.java

```diff
@@ -107,104 +100,80 @@ private void getUser(
       return;
     }
 
-    // We do the actual authorization here not in the User class
-    final boolean authorized = validateRoles(user, allowedRoles);
-
     // if the subject from the login context is already a KsqlPrincipal, use the subject
     // (wrapped inside another DefaultKsqlPrincipal) rather than creating a new one
     final Optional<KsqlPrincipal> ksqlPrincipal = user.getSubject().getPrincipals().stream()
-        .filter(p -> p instanceof KsqlPrincipal)
-        .map(p -> (KsqlPrincipal)p)
+        .filter(KsqlPrincipal.class::isInstance)
+        .map(KsqlPrincipal.class::cast)
         .findFirst();
-    final JaasUser jaasUser = ksqlPrincipal.isPresent()
-        ? new JaasUser(ksqlPrincipal.get(), authorized)
-        : new JaasUser(username, password, authorized);
-    promise.complete(jaasUser);
-  }
-
-  private static boolean validateRoles(final UserIdentity ui, final List<String> allowedRoles) {
-    if (allowedRoles.contains("*")) {
-      // all users allowed
-      return true;
-    }
-
-    final Set<String> userRoles = ui.getSubject().getPrincipals().stream()
-        .map(Principal::getName)
-        .collect(Collectors.toSet());
-    return !CollectionUtils.intersection(userRoles, allowedRoles).isEmpty();
-  }
-
-  @SuppressWarnings("deprecation")
-  static class JaasUser extends io.vertx.ext.auth.AbstractUser implements ApiUser {
-
-    private final DefaultKsqlPrincipal principal;
-    private final boolean authorized;
-
-    JaasUser(
-        final String username,
-        final String password,
-        final boolean authorized
-    ) {
-      this.principal = new JaasPrincipal(
-          Objects.requireNonNull(username, "username"),
-          Objects.requireNonNull(password, "password")
-      );
-      this.authorized = authorized;
-    }
-
-    JaasUser(
-        final KsqlPrincipal principal,
-        final boolean authorized
-    ) {
-      Objects.requireNonNull(principal, "principal");
-      this.principal = principal instanceof DefaultKsqlPrincipal
-          ? (DefaultKsqlPrincipal) principal
-          : new DefaultKsqlPrincipal(principal);
-      this.authorized = authorized;
-    }
-
-    @Override
-    public void doIsPermitted(
-        final String permission,
-        final Handler<AsyncResult<Boolean>> resultHandler
-    ) {
-      resultHandler.handle(Future.succeededFuture(authorized));
-    }
-
-    @Override
-    public JsonObject principal() {
-      throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void setAuthProvider(final AuthProvider authProvider) {
-    }
 
-    @Override
-    public DefaultKsqlPrincipal getPrincipal() {
-      return principal;
-    }
+    final Authorizations authorizations = new AuthorizationsImpl();
+    user.getSubject()
+        .getPrincipals()
+        .forEach(p -> authorizations.add("default", RoleBasedAuthorization.create(p.getName())));
+
+    promisedUser.complete(new ApiUser() {
```

Comment:
Any particular reason to use an anonymous class here?  I'm kinda ambivalent, but figured I'd ask since we had a concrete one previously.

Reply:
not really 🤷  I think this is an artifact of some refactoring I did but I don't remember what